bin/varnishd/cache/cache_http.c

```diff
@@ -821,36 +821,158 @@ http_GetHdrField(const struct http *hp, hdr_t hdr,
 
 /*--------------------------------------------------------------------*/
 
+static ssize_t
+http_parse_uint(const char *b, const char **e)
+{
+	ssize_t u;
+	unsigned n;
+
+	u = 0;
+	if (!vct_isdigit(*b))
+		return (-1);
+	for (; vct_isdigit(*b); b++) {
+		if (u > (SSIZE_MAX / 10))
+			return (-1);
+		u *= 10;
+		n = *b - '0';
+		if (u > (SSIZE_MAX - n))
```

Comment:
to avoid the duplicate checks for SSIZE_MAX, shouldn't we just have
```
n = *b - '0';
if (u > (SSIZE_MAX - n))
    return (-2);
u *= 10;
u += n;
```

Reply:
Did you mean `if (u > (SSIZE_MAX - n) / 10)` ?

Reply:
that is the one I think we do not need if we have `if (u > (SSIZE_MAX - n))`

Reply:
Something for after it has moved to VNUM?